### PR TITLE
fix: Make initializer in OneOnOneResolver non optional - WPB-5815

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5815" title="WPB-5815" target="_blank">Sub-taskWPB-5815</a>  [iOS] Periodically evaluate a single 1:1 conversation
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we make the initializer of the OneOnOneResolver non optional and instead we make the Migrator to be optional. The reason for this change is that sometimes we want to use the OneOnOneResolver with the migrator and sometimes we don't. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — COMMENTED** [review]
You will probably need to update code that was using the fallible initializer (since it's no longer fallible).

File: wire-ios-data-model/Source/Model/User/ZMUser.swift [hunk 1]
```diff
@@ -452,11 +452,7 @@ extension ZMUser: UserConnections {
             case .success:
                 Task {
                     do {
-                        guard let resolver = oneOnOneResolver ?? OneOnOneResolver(syncContext: syncContext) else {
-                            completion(AcceptConnectionError.unableToResolveConversation)
-                            return
-                        }
-
+                        let resolver: OneOnOneResolverInterface = oneOnOneResolver ?? OneOnOneResolver(syncContext: syncContext)
```
Comment (johnxnguyen) [inline wire-ios-data-model/Source/Model/User/ZMUser.swift]: suggestion: delete the explicit type declaration if you can.

File: wire-ios-data-model/Source/Model/User/ZMUser.swift [hunk 2]
```diff
@@ -452,11 +452,7 @@ extension ZMUser: UserConnections {
             case .success:
                 Task {
                     do {
-                        guard let resolver = oneOnOneResolver ?? OneOnOneResolver(syncContext: syncContext) else {
-                            completion(AcceptConnectionError.unableToResolveConversation)
```
Comment (johnxnguyen) [inline wire-ios-data-model/Source/Model/User/ZMUser.swift:456]: suggestion: delete this error if you can.

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios-data-model/Source/Model/User/ZMUser.swift (hunk 1 above)
Comment (caldrian) [inline wire-ios-data-model/Source/Model/User/ZMUser.swift]: Once created, why don't we keep the instance?

File: wire-ios-data-model/Source/Model/User/ZMUser.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/User/ZMUser.swift:456]: Done!

File: wire-ios-data-model/Source/Model/User/ZMUser.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/User/ZMUser.swift]: Done!

**Review by caldrian — APPROVED** [review]

File: wire-ios-data-model/Source/Model/User/ZMUser.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/User/ZMUser.swift]: We discussed it 1:1 I think we can resolve the comment. :)

**Pull request merged**